### DISCUSSION TITLE
Authentication

### DISCUSSION THREAD

**Discussion by almakamberi97** [supabase/supabase#3750] [Questions] [answered]
As they say there are no good or bad question but feel rather dumb asking everybody this?
I'd like to know if there is any choice or any idea of how to implement an email authentication template(the redirected  page) from the verification  link sent in email? 
As known firebase does provide one. 
So how do I built one myself if using dart as my front-end

**Comment by dshukertjr** [comment, marked as answer]
Hi @almakamberi97!

You can use [`supabase_flutter`](https://pub.dev/packages/supabase_flutter) package to handle redirect of the app and email confirmation link click. You can read more on how to implement it by reading though the [quick starter guide](https://supabase.io/docs/guides/with-flutter).
